configuration/src/main/java/com/synopsys/integration/configuration/help/PropertyConfigurationHelpContext.java

```diff
@@ -120,6 +116,12 @@ public void printPropertyErrors(Consumer<String> logger, List<Property> knownPro
             });
     }
 
+    private List<Property> sortPropertiesByKey(List<Property> knownProperties) {
```

Comment:
Wait we already have a sorted list of properties? Why even sort the map at all then? I'd understand if we never sorted the known properties, but if we already sort them (or occasionally sort them twice?), why not just use them then.

Reply:
I feel like it's still the same question as posed in the poll.  We could sort the list and then let that dictate the order we query the map, or just sort the map.  Also printCurrentValues doesn't currently need to consume the list of properties so the prior approach would be adding an extra dependency.

Reply:
I still think the sorting of the map is to be avoided if possible. Even if it is preferred over accessing with the sorted keys. If we can avoid it altogether I would prefer that.